### PR TITLE
Fixed #24609 --  Made ConcatPair on MySQL use CONCAT_WS()

### PR DESCRIPTION
**Pull request by adamchainz**
https://code.djangoproject.com/ticket/24609

### CODE REVIEW

File: django/db/models/functions.py [hunk 1]
```diff
@@ -47,7 +47,8 @@ def as_sqlite(self, compiler, connection):
         return super(ConcatPair, self).as_sql(compiler, connection)
 
     def as_mysql(self, compiler, connection):
-        self.coalesce()
+        self.function = 'CONCAT_WS'
```
Comment (jarshwah) [inline django/db/models/functions.py]: I would like a brief comment here mentioning the use of concat_ws due to the handling of nulls.

File: django/db/models/functions.py [hunk 2]
```diff
@@ -47,7 +47,8 @@ def as_sqlite(self, compiler, connection):
...
+        self.template = '%(function)s("", %(expressions)s)'
```
Comment (jarshwah) [inline django/db/models/functions.py]: Does the use of double quotes for the separator argument matter here? I'm used to most string values having to use single quotes, but I'm not a mysql user so I'm not entirely sure it matters. I did a quick search and it appears that mysql supports double or single quotes for string literals.

**Comment by jarshwah** [comment]
Other than a comment mentioning the use of CONCAT_WS this looks good to me. I didn't like having to wrap concat with coalesce, so this work around is nice.

File: django/db/models/functions.py (hunk 2 above)
Comment (adamchainz) [inline django/db/models/functions.py]: Good point, it should be using single quotes. It supports both for string literals until you turn on ANSI_QUOTES sql_mode, which I don't have locally but others might.

File: django/db/models/functions.py (hunk 1 above)
Comment (adamchainz) [inline django/db/models/functions.py]: Sure.

**Comment by timgraham** [comment]
merged in 9d0c600d8d6359107cdbef5f0c30757321da09f6, thanks!

**Pull request closed without merging**